pkg/controllers/managementuser/clusterauthtoken/common/user.go

```diff
@@ -21,16 +23,45 @@ func NewClusterAuthToken(token *managementv3.Token, hashedValue string) (*cluste
 		TypeMeta: metav1.TypeMeta{
 			Kind: "ClusterAuthToken",
 		},
-		UserName:      token.UserID,
-		SecretKeyHash: hashedValue,
-		ExpiresAt:     token.ExpiresAt,
-		Enabled:       tokenEnabled,
+		UserName:  token.UserID,
+		ExpiresAt: token.ExpiresAt,
+		Enabled:   tokenEnabled,
 	}
 	return result, nil
 }
 
-// VerifyClusterAuthToken verifies that a provided secret key is valid for the given clusterAuthToken.
-func VerifyClusterAuthToken(secretKey string, clusterAuthToken *clusterv3.ClusterAuthToken) error {
+// NewClusterAuthSecret creates a new secret from the given token and its hash value
+// The cluster auth token is managed separately.
+// Does not create the secret in the remote cluster.
+func NewClusterAuthSecret(token *managementv3.Token, hashedValue string) (*corev1.Secret, error) {
+	result := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ClusterAuthSecretName(token.ObjectMeta.Name),
+		},
```

Comment:
Q: Do we want to set the `ownerReference` to the corresponding ClusterAuthToken and let k8s to do the cleanup instead of/in addition to the lifetime controller event?

Reply:
Would be nice. Requires restructuring the code ... Oh, :bulb: ... Owner-ref could be added to secret in memory, after the CAT is created (because we need the CAT UID, IIRC). If we want to add this info here, we have restructure to make this after the CAT is created. Currently both resources are made in memory together and then created in sequence.

Thinking...

Reply:
commit f350585d37d2d26e9862b9e69a649e2e765d3ef1